velox/exec/tests/HashJoinTest.cpp

```diff
@@ -586,6 +586,107 @@ TEST_F(HashJoinTest, semiJoinWithFilter) {
       "SELECT t.* FROM t WHERE EXISTS (SELECT u0, u1 FROM u WHERE t0 = u0 AND t1 <> u1)");
 }
 
+TEST_F(HashJoinTest, rightSemiJoin) {
+  auto leftVectors = makeRowVector({
+      makeFlatVector<int32_t>(
+          1'234, [](auto row) { return row % 11; }, nullEvery(13)),
+      makeFlatVector<int32_t>(1'234, [](auto row) { return row; }),
+  });
+
+  auto rightVectors = makeRowVector({
+      makeFlatVector<int32_t>(
+          123, [](auto row) { return row % 5; }, nullEvery(7)),
+  });
+
+  createDuckDbTable("t", {leftVectors});
+  createDuckDbTable("u", {rightVectors});
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  auto op =
+      PlanBuilder(planNodeIdGenerator)
+          .values({rightVectors})
+          .hashJoin(
+              {"c0"},
+              {"c0"},
+              PlanBuilder(planNodeIdGenerator).values({leftVectors}).planNode(),
+              "",
+              {"c1"},
+              core::JoinType::kRightSemi)
+          .planNode();
+
+  assertQuery(op, "SELECT t.c1 FROM t WHERE t.c0 IN (SELECT c0 FROM u)");
```

Comment:
This example is confusing because this reads like a left semi join. Is there an equivalent query we can write that looks like a right semi-join ?

Reply:
I updated the description. There is no SQL statement that purely represents a right semi join. EXISTS/IN queries can be implemented using a left or a right semi join, whichever keeps the build size small.